ref: `remote modify` `gdrive_acknowledge_abuse` and `profile` (#3795)
iterative/dvc.org: content/docs/user-guide/setup-google-drive-remote.md

example when trying to `dvc push` tracked data for the first time, DVC will
prompt you to visit a special Google authentication web page. There you'll need
to sign into a Google account with the needed access to the GDrive
[URL](#url-format) in question. The [auth process] will ask you to grant DVC the
necessary permissions, and produce a verification code needed for DVC to
complete the connection. On success, the necessary credentials will be cached

> globally ([see `gdrive_user_credentials_file`]) and used automatically next time

Reviewer: let's put a proper example and see also ...

Reviewer 2: Not exactly sure what you mean. Proper example = `~/Library/Caches/...`? If so please commit:

```suggestion
globally, for example in
`~/Library/Caches/pydrive2fs/{gdrive_client_id}/default.json` for Linux
([see `gdrive_user_credentials_file`]), and used automatically next time
```

Author: @jorgeorpinel That's incorrect for linux. It is more like "~/.cache/pydrife2fs/{gdrive_client_id}/default.json", as indicated by the table
<img width="529" alt="Screenshot 2022-07-29 at 20 53 42" src="https://user-images.githubusercontent.com/5367102/181817112-265cacb1-4cc9-4776-b621-04c6e33e2211.png">

Reviewer 2: Right, I meant macOS. Changed. Committing...

DVC needs them.

<admin type="warn">

In order to prevent unauthorized access to your Google Drive, **do not share
these credentials with others**. Each team member should go through this process
individually.

</admin>

> If you use multiple GDrive remotes, by default they will share the same cached

Reviewer: they won't, only if it's only the same client also

credentials. To isolate them, you can use custom profile names for different
remotes:

```

> $ dvc remote modify --local myremote profile 'myuser'

Reviewer: myprofile

```

You can also overwrite the cached credentials file location completely, for
example to have it inside your project directory:

```dvc
$ dvc remote modify myremote --local \
      gdrive_user_credentials_file .dvc/tmp/myremote-credentials.json
```

<admin type="warn">

If the file is inside a Git repo, **do not commit it** to Git in order to
prevent unauthorized access to your Google Drive. Add it to `.gitignore` to be

> sure.

Reviewer 2: Instead of giving an example where the custom credentials file location is inside the repo and then having to warn about not putting it in the repo, should the example be in the home directory? We can still warn against putting it in the repo though.

Reviewer: yes, it makes sense to me

Reviewer 2: > should the (`profile`) example be in the home directory 

Done.

</admin>

To change the user you have authenticated with or for troubleshooting misc.
token errors, you can remove the user credentials file and authorize again.

Alternatively, a `GDRIVE_CREDENTIALS_DATA` can be set to pass user credentials
in CI/CD systems, production setup, read-only file systems, etc. The content of

> this variable should be a JSON string with the same content as credentials file

Reviewer: why touching this? is it related?

Reviewer 2: I guess not. For a moment I thought this PR also introduced `gdrive_user_credentials_file` but no. Reverted.

[auth process](https://developers.google.com/drive/api/v2/about-auth) will ask
you to grant DVC the necessary permissions, and produce a verification code
needed for DVC to complete the connection. On success, the necessary credentials

> will be cached and they will be used automatically next time you run DVC.

Reviewer 2: ```suggestion
will be cached (see `gdrive_user_credentials_file` in the [GDrive remote
params]) and used automatically next time DVC needs them.

[gdrive remote params]:
  https://dvc.org/doc/command-reference/remote/modify#google-drive
```

⚠️ In order to prevent unauthorized access to your Google Drive, **do not share
these credentials with others**. Each team member should go through this process
individually.

If you use multiple GDrive remotes, by default they will be sharing the same

> cached credentials file. It can be overridden with the
> `gdrive_user_credentials_file` parameter:

Reviewer 2: ```suggestion
cached credentials. This can be overridden with
`dvc remote modify`:
```

[auth process](https://developers.google.com/drive/api/v2/about-auth) will ask
you to grant DVC the necessary permissions, and produce a verification code
needed for DVC to complete the connection. On success, the necessary credentials
will be cached (see `gdrive_user_credentials_file` in the [GDrive remote
params]) and used automatically next time DVC needs them.

[gdrive remote params]:
  https://dvc.org/doc/command-reference/remote/modify#google-drive

⚠️ In order to prevent unauthorized access to your Google Drive, **do not share
these credentials with others**. Each team member should go through this process
individually.

If you use multiple GDrive remotes, by default they will be sharing the same

> cached credentials. This can be overridden with
> `dvc remote modify`:

Reviewer 2: ```suggestion
cached credentials. This can be overridden with `dvc remote modify`:
```

If you wish to change the user you have authenticated with, or for

> troubleshooting misc. token errors, simply remove the user credentials JSON file

Reviewer 2: ```suggestion
troubleshooting misc. token errors, you can remove the user credentials file
```

> and authorize again. You might also want to use `profile` parameter to make DVC
> cache your credentials per-`profile` instead of per-`client_id`:

Reviewer 2: ```suggestion
and authorize again. Alternatively you can use a custom `profile`-based
credential cache (instead of the default, `gdrive_client_id`-based one):
```

it with other people.

If you wish to change the user you have authenticated with, or for

> troubleshooting misc. token errors, you can remove the user credentials file
> and authorize again. Alternatively you can use a custom `profile`-based
> credential cache (instead of the default, `gdrive_client_id`-based one):

Reviewer 2: Format
```suggestion
troubleshooting misc. token errors, you can remove the user credentials file and
authorize again. Alternatively you can use a custom `profile`-based credential
cache (instead of the default, `gdrive_client_id`-based one):
```
